app/code/Magento/Ui/view/base/web/js/grid/data-storage.js

```diff
@@ -267,10 +267,12 @@ define([
          * @param {Object} data - Response data.
          */
         onRequestComplete: function (params, data) {
-            this.updateData(data.items);
+            if (data) {
```

Comment:
Be sure that empty response also will be added to request cache to prevent possible additional requests.
You can check my suggestion.
```suggestion
            if (data) {
                this.updateData(data.items);
            } else {
                data = {};
            }
```